haystack/database/base.py

```diff
@@ -38,9 +38,14 @@ class Document(BaseModel):
     question: Optional[str] = Field(None, description="Question text for FAQs.")
     query_score: Optional[float] = Field(None, description="Elasticsearch query score for a retrieved document")
     meta: Optional[Dict[str, Any]] = Field(None, description="")
+    tags: Optional[Dict[str, Any]] = Field(None, description="Tags that allow filtering of the data")
```

Comment:
Would it be better to have a separate `tags` field than using the existing `meta`? IMO, basic filtering by key, value(s) pairs could work with the `meta` field.

Reply:
I've looked at the code of other datastores, the document filtering mentions "tags" as a data structure, it looks like it deserves its own space. Also, I can see how meta could change its type into a list of dicts, and tags would probably stay a dictionary.